src/js/components/utils/useFetchIssueIfMissing.tsx

```diff
@@ -0,0 +1,29 @@
+import { useCallback, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { AppState, ThunkDispatch } from '@/js/store';
+import { fetchObject } from '@/js/store/actions';
+import { selectIssue } from '@/js/store/githubIssues/selectors';
+import { OBJECT_TYPES } from '@/js/utils/constants';
+
+export default ({ id }: { id: string }) => {
+  const dispatch = useDispatch<ThunkDispatch>();
+  const selectIssueWithArgs = useCallback(selectIssue, []);
+  const issue = useSelector((state: AppState) =>
+    selectIssueWithArgs(state, id),
+  );
+
+  useEffect(() => {
+    if (id && issue === undefined) {
+      // Fetch issue from API
+      dispatch(
+        fetchObject({
+          objectType: OBJECT_TYPES.GITHUB_ISSUE,
+          filters: { id },
+        }),
+      );
```

Comment:
@oluoluoxenfree I went ahead and moved these attached issues into the Redux store, so we don't have to re-fetch them every time we navigate to an Epic/Task again.